src/PromiseCore.php

```diff
@@ -192,7 +192,12 @@ public function reject(Exception $exception)
     {
         $this->exception = $exception;
         $this->state = Promise::REJECTED;
-        $this->exception = $this->call($this->onRejected, $this->exception);
+
+        try {
+            $this->call($this->onRejected, $this->exception);
+        } catch (Exception $exception) {
+            $this->exception = $exception;
```

Comment:
I guess the callback should handle the exception or rethrow it. But @joelwurtz is the promise master in da house.